src/lightning-address.ts

```diff
@@ -24,9 +25,9 @@ export default class LightningAddress {
   username: string | undefined;
   domain: string | undefined;
   pubkey: string | undefined;
-  lnurlpData: Record<string, any>;
-  keysendData: Record<string, any>;
-  nostrData: Record<string, any>;
+  lnurlpData: LnUrlPayResponse | undefined;
```

Comment:
can that still be undefined if we default to `{}` in the constructor?

Reply:
@bumi we cannot set it to `{}` now since the field is correctly typed. It needs the properties of the object to be set.

What is the benefit of doing this? there is no data available to use and apps that do not call fetch will still break. Is the issue that the user has to check if lnurlPData is set before using it, right?

We could make these fields private and then expose public getters that always return a non-nullable version, but throw an exception if the underlying variable is undefined (because fetch was not called).

I think the best solution would be for LightningAddress.fetch() to return a new object which we know has the values set, rather than always interacting with the original lightning address object. This would require a refactor though.

Reply:
Also, my idea above doesn't really work anyway, because we use a single function to fetch 3 different pieces of data, and there's no guarantee that all 3 will exist.